crypto/txscript/src/lib.rs

```diff
@@ -94,8 +100,72 @@ fn parse_script<T: VerifiableTransaction, Reused: SigHashReusedValues>(
     script.iter().batching(|it| deserialize_next_opcode(it))
 }
 
+/// Determines the exact number of signature operations executed in a transaction input
+/// by simulating the script execution. Takes into account conditional branches and only
+/// counts signature operations that are actually executed.
+///
+/// Example of how counts differ:
+/// ```text
+/// IF
+///     CHECKSIG        // 1 sig op if true branch taken
+/// ELSE
+///     CHECKSIG        // 3 sig ops if false branch taken
+///     CHECKSIG
+///     CHECKSIG
+/// ENDIF
+/// ```
+/// `get_sig_op_upper_bound` would return 4, while this function returns 1 or 3
+/// depending on which branch is actually executed.
+///
+/// This function should be used:
+/// - After the runtime signature operation counting hardfork activation
+/// - When exact sig op counts are needed for fee calculation
+/// - For accurate validation of sig op limits
+/// - When working with scripts that have conditional logic
+///
+/// # Arguments
+/// * `tx` - The transaction containing the input to analyze
+/// * `input_idx` - Index of the input to analyze
+/// * `kip10_enabled` - Whether KIP-10 features are enabled
+///
+/// # Returns
+/// * `Ok(u8)` - The exact number of signature operations executed
+/// * `Err(TxScriptError)` - If script execution fails or input index is invalid
+pub fn get_sig_op_count<T: VerifiableTransaction>(tx: &T, input_idx: usize, kip10_enabled: bool) -> Result<u8, TxScriptError> {
+    let sig_cache = Cache::new(10_000);
```

Comment:
although `get_sig_op_count` is not used in the current codebase, it seem redundant to actually allocate a cache here. I think `size` should be `0` here (there are no expected cache hits).